core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/plangen/LogicalPlan.java

```diff
@@ -117,20 +117,21 @@ public Schema getOperatorOutputSchema(String operatorID) throws PlanGenException
     /**
      * Updates the current plan and fetch the schema from an operator
      * @param operatorID, the ID of an operator
+     * @param operatorInputSchemaMap Map of operators to their input schemas
      * @return Schema, which includes the attributes setting of the operator
      */
-    public Schema getOperatorOutputSchema(String operatorID, Map<String, List<Schema>> inputSchemas)
+    public Optional<Schema> getOperatorOutputSchema(String operatorID, Map<String, List<Schema>> operatorInputSchemaMap)
             throws PlanGenException, DataflowException {
 
         IOperator currentOperator = operatorObjectMap.get(operatorID);
-        Schema outputSchema = null;
+        Optional<Schema> outputSchema = null;
         if (currentOperator instanceof ISourceOperator) {
-            outputSchema = currentOperator.transformToOutputSchema();
-        } else if (inputSchemas.containsKey(operatorID)) {
-            List<Schema> inputSchema = inputSchemas.get(operatorID);
+            outputSchema = Optional.ofNullable(currentOperator.transformToOutputSchema());
+        } else if (operatorInputSchemaMap.containsKey(operatorID)) {
+            List<Schema> inputSchemaList = operatorInputSchemaMap.get(operatorID);
             try {
-                outputSchema = currentOperator.transformToOutputSchema(
-                        inputSchema.toArray(new Schema[inputSchema.size()]));
+                outputSchema = Optional.ofNullable(currentOperator.transformToOutputSchema(
```

Comment:
@avinash0161 the exception problem is still not solved. When an exception occurs, it still returns `null`, rather than something like `Optional.empty()`